web/packages/shared/components/UnifiedResources/UnifiedResources.tsx

```diff
@@ -291,7 +279,17 @@ export function UnifiedResources(props: UnifiedResourcesProps) {
     });
     setSelectedResources([]);
     setUpdatePinnedResources(makeEmptyAttempt());
-    updateUnifiedResourcesPreferences({ defaultTab: value });
+    updateUnifiedResourcesPreferences({
+      ...unifiedResourcePreferences,
+      defaultTab: value,
+    });
+  };
+
+  const selectViewMode = (viewMode: UnifiedViewModePreference) => {
+    updateUnifiedResourcesPreferences({
+      ...unifiedResourcePreferences,
+      viewMode,
+    });
```

Comment:
This is great 👏 
However, now we have some inconsistency: `viewMode` state is kept outside, but information if pinned tab is active is kept in params and sent externally via `updateUnifiedResourcesPreferences`.
It seems to me that it would be good to have a single way of updating both properties. I made an attempt to refactor this, you can check out the draft https://github.com/gravitational/teleport/pull/34394.
It's not perfect, but we should refactor `useUrlFiltering` first.
I also improved naming and fixed imports (`shared` and `teleterm` shouldn't import from `teleport`).

Reply:
Agreed, though I think we can do the refactoring after this is merged since this PR is already large enough and this is just refactoring which doesn't affect UX.

Reply:
Agreed, though I think we should do the refactoring after this is merged since this PR is already large enough and this is just refactoring which doesn't affect UX.

Reply:
Agreed, though I think we should do the refactoring after this is merged since this PR is already large enough and this doesn't affect UX.